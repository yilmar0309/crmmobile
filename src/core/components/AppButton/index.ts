export {AppButton, AppButtonVariant, AppButtonSizeVariant} from './AppButton';
