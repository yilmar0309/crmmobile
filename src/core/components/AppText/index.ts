export {AppText, AppTextVariant} from './AppText';
