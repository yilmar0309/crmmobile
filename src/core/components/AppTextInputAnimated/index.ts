export {AppTextInputAnimated, TextInputState} from './AppTextInputAnimated';
