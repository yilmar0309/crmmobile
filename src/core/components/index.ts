export {AppText, AppTextVariant} from './AppText';
export {AppButton, AppButtonVariant, AppButtonSizeVariant} from './AppButton';
export {AppTextInputAnimated, TextInputState} from './AppTextInputAnimated';
export {AppContainer} from './AppContainer';
export {AppSpacer} from './AppSpacer';
export {AppHeaderLeftBack} from './AppHeaderLeftBack';
export {AppShowEmptyList} from './AppShowEmptyList';
export {AppLoading} from './AppLoading';
export {AppImage} from './AppImage';
export {AppLoadingList} from './AppLoadingList';
