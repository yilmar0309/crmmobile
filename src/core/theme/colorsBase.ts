export const colorsBase = {
  WHITE: '#FFFFFF',
  BLACK: '#000000',
  GRAY: '#F8F8F8',
};
