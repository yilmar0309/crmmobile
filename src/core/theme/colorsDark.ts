export const colorsDark = {
  BACKGROUND: '#2C3E50',
  ON_BACKGROUND: '#E1E2E1',

  // Background primary color
  PRIMARY_COLOR: '#4d9ecf',
  ON_PRIMARY: '#FFFFFF',
  SECONDARY_COLOR: '#356ab0',

  // Background
  BACKGROUND_SCREEN_COLOR: '#2C3E50',

  // SURFACE
  SURFACE_1: '#34495E',

  // CONTENT
  CONTENT_PRIMARY: '#E1E2E1',
  CONTENT_SECONDARY: '#AAB0B6',

  // Text color
  PRIMARY_TEXT_COLOR: '#E1E2E1',
  SECONDARY_TEXT_COLOR: '#AAB0B6',
  DISABLED_TEXT_COLOR: '#52555A',
  TEXT_SIMPLE_COLOR: '#34495E',

  // TextInput color
  BACKGROUND_TEXT_INPUT_COLOR: '#34495E',

  // Button
  BACKGROUND_BUTTON_COLOR: '#2C3E50',
  BUTTON_TEXT_CONTAINED_COLOR: '#FFFFFF',
  BUTTON_TEXT_OUTLINED_COLOR: '#67b7ed',
  BUTTON_TEXT_COLOR: '#FFFFFF',
  BUTTON_BORDER_COLOR: '#67b7ed',
  BUTTON_DISABLED_COLOR: '#AAB0B6',
  BUTTON_DISABLED_TEXT_COLOR: '#CDD1D0',

  // Bottom Tab
  BOTTOM_TAB_COLOR: '#E1E2E1',

  // GRAY
  GRAY_01: '#34495E',
  GRAY_02: '#273746',
  GRAY_03: '#1B2C3B',
  GRAY_04: '#22313F',
  GRAY_05: '#293B4D',
  GRAY_06: '#999999',

  // Alert color
  SUCCESS_COLOR: '#2D7D46',
  WARNING_COLOR: '#C97B00',
  ERROR_COLOR: '#D0342C',
  INFO_COLOR: '#8D8B92',

  WHATSAPP: '#1E7D47',
  FACEBOOK: '#1D508D',
  INSTAGRAM: '#AD2671',
  TIKTOK: '#181818',

  // Basic color
  WHITE: '#FFFFFF',
  BLACK: '#121212',

  OUTLINE: '#2C313C',

  // SENTIMENT
  SENTIMENT_SUCCESS_BACKGROUND: '#1B4026',
  SENTIMENT_SUCCESS: '#2D7D46',
  SENTIMENT_INFO_BACKGROUND: '#2C3E50',
  SENTIMENT_INFO: '#1E2620',
  SENTIMENT_WARNING_BACKGROUND: '#332B1A',
  SENTIMENT_WARNING: '#C97B00',
  SENTIMENT_NEGATIVE_ERROR_BACKGROUND: '#401311',
  SENTIMENT_NEGATIVE_COLOR: '#D0342C',
  SENTIMENT_TEXT: '#1E2620',
  SENTIMENT_POSITIVE_BG: '#1B4026',
  SENTIMENT_WARNING_BACKGROUND_LOCK: '#664D1A',
  SENTIMENT_WARNING_ON_WARNING_BG: '#3A2800',

  // MAPS
  COLOR_MAP_LOSSES: '#FF6B6B',
  COLOR_MAP_FOUND: '#4ECDC4',
  COLOR_MAP_ADOPTION: '#FFD93D',
};
