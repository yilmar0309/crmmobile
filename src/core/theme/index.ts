export {colorsBase} from './colorsBase';
export {colorsLight} from './colorsLight';
export {colorsDark} from './colorsDark';
