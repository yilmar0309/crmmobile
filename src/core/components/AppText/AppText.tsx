/* eslint-disable react-native/no-unused-styles */
import React from 'react';
import {StyleSheet, Text, TextProps, useColorScheme} from 'react-native';
import {colorsDark, colorsLight} from '@/core/theme';

export enum AppTextVariant {
  h1 = 'h1',
  h2 = 'h2',
  h3 = 'h3',
  h4 = 'h4',
  h5 = 'h5',
  h6 = 'h6',
  subtitle1 = 'subtitle1',
  subtitle2 = 'subtitle2',
  body1 = 'body1',
  body2 = 'body2',
  button = 'button',
  caption = 'caption',
  overline = 'overline',
}

interface AppTextProps extends TextProps {
  variant: AppTextVariant;
  children: React.ReactNode;
  color?: string;
}

export const AppText = ({
  variant,
  children,
  style,
  color,
  ...props
}: AppTextProps) => {
  const isDarkMode = useColorScheme() === 'dark';
  const textColor = isDarkMode
    ? colorsDark.PRIMARY_TEXT_COLOR
    : colorsLight.PRIMARY_TEXT_COLOR;
  return (
    <Text
      style={[styles[variant], {color: color || textColor}, style]}
      {...props}>
      {children}
    </Text>
  );
};

const styles = StyleSheet.create({
  h1: {
    fontWeight: 'bold',
    fontSize: 96,
    letterSpacing: -1.5,
  },
  h2: {
    fontWeight: '600',
    fontSize: 60,
    letterSpacing: -0.5,
  },
  h3: {
    fontWeight: '400',
    fontSize: 48,
  },
  h4: {
    fontWeight: '400',
    fontSize: 34,
    letterSpacing: 0.25,
  },
  h5: {
    fontWeight: '400',
    fontSize: 24,
  },
  h6: {
    fontWeight: '600',
    fontSize: 20,
    letterSpacing: 0.15,
  },
  subtitle1: {
    fontWeight: '400',
    fontSize: 16,
    letterSpacing: 0.15,
  },
  subtitle2: {
    fontWeight: '600',
    fontSize: 14,
    letterSpacing: 0.1,
  },
  body1: {
    fontWeight: '400',
    fontSize: 16,
    letterSpacing: 0.5,
  },
  body2: {
    fontWeight: '400',
    fontSize: 14,
    letterSpacing: 0.25,
  },
  button: {
    fontWeight: '600',
    fontSize: 14,
    letterSpacing: 1.25,
  },
  caption: {
    fontWeight: '400',
    fontSize: 12,
    letterSpacing: 0.4,
  },
  overline: {
    fontWeight: '400',
    fontSize: 10,
    letterSpacing: 1.5,
  },
});
