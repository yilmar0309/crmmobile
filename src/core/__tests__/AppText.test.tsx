import React from 'react';
import {StyleSheet} from 'react-native';
import {render} from '@testing-library/react-native';
import {AppText, AppTextVariant} from '../components/AppText';
import {colorsDark, colorsLight} from '../theme';
const mockedAppearance = require('react-native/Libraries/Utilities/Appearance');

describe('AppText', () => {
  beforeEach(() => {
    mockedAppearance.setColorScheme('dark');
  });

  afterEach(() => {
    mockedAppearance.setColorScheme('light');
  });

  const testText = 'Test Text';

  const testVariants = Object.values(AppTextVariant);

  testVariants.forEach(variant => {
    it(`renders correctly with variant ${variant} in light mode`, () => {
      mockedAppearance.setColorScheme('light');
      const {getByText, toJSON} = render(
        <AppText variant={variant}>{testText}</AppText>,
      );
      const element = getByText(testText);
      expect(element).toBeTruthy();

      // Flattening the style to handle arrays of styles
      const flattenedStyle = StyleSheet.flatten(element.props.style);

      expect(flattenedStyle.color).toBe(colorsLight.PRIMARY_TEXT_COLOR);
      expect(toJSON()).toMatchSnapshot();
    });

    it(`renders correctly with variant ${variant} in dark mode`, () => {
      const {getByText} = render(
        <AppText variant={variant}>{testText}</AppText>,
      );
      const element = getByText(testText);
      expect(element).toBeTruthy();

      // Flattening the style to handle arrays of styles
      const flattenedStyle = StyleSheet.flatten(element.props.style);

      expect(flattenedStyle.color).toBe(colorsDark.PRIMARY_TEXT_COLOR);
    });
  });

  it('applies custom color when provided', () => {
    const customColor = '#123456';
    const {getByText} = render(
      <AppText variant={AppTextVariant.body1} color={customColor}>
        {testText}
      </AppText>,
    );
    const element = getByText(testText);
    // Flattening the style to handle arrays of styles
    const flattenedStyle = StyleSheet.flatten(element.props.style);
    expect(flattenedStyle.color).toBe(customColor);
  });
});
