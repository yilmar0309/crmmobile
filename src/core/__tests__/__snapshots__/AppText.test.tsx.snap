// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppText renders correctly with variant body1 in light mode 1`] = `
<Text
  style={
    [
      {
        "fontSize": 16,
        "fontWeight": "400",
        "letterSpacing": 0.5,
      },
      {
        "color": "#263238",
      },
      undefined,
    ]
  }
>
  Test Text
</Text>
`;

exports[`AppText renders correctly with variant body2 in light mode 1`] = `
<Text
  style={
    [
      {
        "fontSize": 14,
        "fontWeight": "400",
        "letterSpacing": 0.25,
      },
      {
        "color": "#263238",
      },
      undefined,
    ]
  }
>
  Test Text
</Text>
`;

exports[`AppText renders correctly with variant button in light mode 1`] = `
<Text
  style={
    [
      {
        "fontSize": 14,
        "fontWeight": "600",
        "letterSpacing": 1.25,
      },
      {
        "color": "#263238",
      },
      undefined,
    ]
  }
>
  Test Text
</Text>
`;

exports[`AppText renders correctly with variant caption in light mode 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12,
        "fontWeight": "400",
        "letterSpacing": 0.4,
      },
      {
        "color": "#263238",
      },
      undefined,
    ]
  }
>
  Test Text
</Text>
`;

exports[`AppText renders correctly with variant h1 in light mode 1`] = `
<Text
  style={
    [
      {
        "fontSize": 96,
        "fontWeight": "bold",
        "letterSpacing": -1.5,
      },
      {
        "color": "#263238",
      },
      undefined,
    ]
  }
>
  Test Text
</Text>
`;

exports[`AppText renders correctly with variant h2 in light mode 1`] = `
<Text
  style={
    [
      {
        "fontSize": 60,
        "fontWeight": "600",
        "letterSpacing": -0.5,
      },
      {
        "color": "#263238",
      },
      undefined,
    ]
  }
>
  Test Text
</Text>
`;

exports[`AppText renders correctly with variant h3 in light mode 1`] = `
<Text
  style={
    [
      {
        "fontSize": 48,
        "fontWeight": "400",
      },
      {
        "color": "#263238",
      },
      undefined,
    ]
  }
>
  Test Text
</Text>
`;

exports[`AppText renders correctly with variant h4 in light mode 1`] = `
<Text
  style={
    [
      {
        "fontSize": 34,
        "fontWeight": "400",
        "letterSpacing": 0.25,
      },
      {
        "color": "#263238",
      },
      undefined,
    ]
  }
>
  Test Text
</Text>
`;

exports[`AppText renders correctly with variant h5 in light mode 1`] = `
<Text
  style={
    [
      {
        "fontSize": 24,
        "fontWeight": "400",
      },
      {
        "color": "#263238",
      },
      undefined,
    ]
  }
>
  Test Text
</Text>
`;

exports[`AppText renders correctly with variant h6 in light mode 1`] = `
<Text
  style={
    [
      {
        "fontSize": 20,
        "fontWeight": "600",
        "letterSpacing": 0.15,
      },
      {
        "color": "#263238",
      },
      undefined,
    ]
  }
>
  Test Text
</Text>
`;

exports[`AppText renders correctly with variant overline in light mode 1`] = `
<Text
  style={
    [
      {
        "fontSize": 10,
        "fontWeight": "400",
        "letterSpacing": 1.5,
      },
      {
        "color": "#263238",
      },
      undefined,
    ]
  }
>
  Test Text
</Text>
`;

exports[`AppText renders correctly with variant subtitle1 in light mode 1`] = `
<Text
  style={
    [
      {
        "fontSize": 16,
        "fontWeight": "400",
        "letterSpacing": 0.15,
      },
      {
        "color": "#263238",
      },
      undefined,
    ]
  }
>
  Test Text
</Text>
`;

exports[`AppText renders correctly with variant subtitle2 in light mode 1`] = `
<Text
  style={
    [
      {
        "fontSize": 14,
        "fontWeight": "600",
        "letterSpacing": 0.1,
      },
      {
        "color": "#263238",
      },
      undefined,
    ]
  }
>
  Test Text
</Text>
`;
